dist/testbed-support/makefile.iotlab.single.inc.mk

```diff
@@ -0,0 +1,190 @@
+#    * Select nodes by architucture using the board-archi mapping
+#    * Nodes for current server in 'auto'
+ifneq (,$(filter auto auto-ssh,$(IOTLAB_NODE)))
+  ifeq (,$(IOTLAB_ARCHI))
+    $(error Could not find 'archi' for $(BOARD), update mapping in $(lastword $(MAKEFILE_LIST)))
+  endif
+
+  _NODES_DEPLOYED = $(shell iotlab-experiment --jmespath='deploymentresults."0"' --format='" ".join' get $(_IOTLAB_EXP_ID) --print)
+  ifeq (auto,$(IOTLAB_NODE))
+    _NODES_DEPLOYED := $(filter %.$(shell hostname).iot-lab.info, $(_NODES_DEPLOYED))
+  endif
+  _NODES_FOR_BOARD = $(shell iotlab-experiment --jmespath="items[?archi=='$(IOTLAB_ARCHI)'].network_address" --format='" ".join' get $(_IOTLAB_EXP_ID) --resources)
+
+  _IOTLAB_NODE := $(word $(IOTLAB_NODE_AUTO_NUM),$(filter $(_NODES_DEPLOYED),$(_NODES_FOR_BOARD)))
+  ifeq (auto,$(IOTLAB_NODE))
+    override IOTLAB_NODE := $(firstword $(subst ., ,$(_IOTLAB_NODE)))
+  else
+    override IOTLAB_NODE := $(_IOTLAB_NODE)
+  endif
+
+  ifeq (,$(IOTLAB_NODE))
+    $(error Could not automatically find a node for BOARD=$(BOARD))
+  endif
+  override IOTLAB_NODE := $(patsubst node-%,%,$(IOTLAB_NODE))
+endif
+
+
+# If the IOTLAB_NODE format is:
+# * 'type-num.iot.lab.info' assume it is run from your computer
+# * 'type-num' assume it is run from iot-lab frontend
+ifneq (,$(filter %.iot-lab.info, $(IOTLAB_NODE)))
+  _IOTLAB_ON_FRONTEND =
+else
+  _IOTLAB_ON_FRONTEND = 1
+endif
+# Work with node url without 'node-'
+override IOTLAB_NODE := $(patsubst node-%,%,$(IOTLAB_NODE))
+
+
+# Create node list and optionnaly frontend url
+ifeq (,$(_IOTLAB_ON_FRONTEND))
+  # m3-380.grenoble.iot-lab.info    -> grenoble,m3,380
+  # a8-1.grenoble.iot-lab.info      -> grenoble,a8,1
+  _NODELIST_SED := 's/\([^.]*\)-\([^.]*\).\([^.]*\).*/\3,\1,\2/'
+  _IOTLAB_NODELIST := --list $(shell echo '$(IOTLAB_NODE)' | sed $(_NODELIST_SED))
+
+  # Remove the node type-number part
+  _IOTLAB_SERVER := $(shell echo '$(IOTLAB_NODE)' | sed 's/[^.]*.//')
+  _IOTLAB_AUTHORITY = $(IOTLAB_USER)@$(_IOTLAB_SERVER)
+else
+  # m3-380    -> $(hostname),m3,380
+  # a8-1      -> $(hostname),a8,1
+  _NODELIST_SED := 's/\([^.]*\)-\([^.]*\)/$(shell hostname),\1,\2/'
+  _IOTLAB_NODELIST := --list $(shell echo '$(IOTLAB_NODE)' | sed $(_NODELIST_SED))
+endif
+
+
+# Display value of IOTLAB_NODE, useful to get the value calculated when using
+# IOTLAB_NODE=auto or auto-ssh
+.PHONY: info-iotlab-node
+info-iotlab-node:
+	@echo $(IOTLAB_NODE)
+
+
+# Configure FLASHER, RESET, TERMPROG depending on BOARD and if on frontend
+
+# Command to check if 'stdin' is 0. Cannot use 'cmp - <(echo 0)' without bash shell
+_STDIN_EQ_0 = grep 0
+
+ifneq (iotlab-a8-m3,$(BOARD))
+
+  # M3 and wsn430 nodes
+  FLASHER     = iotlab-node
+  RESET       = iotlab-node
+  _NODE_FMT   = --jmespath='keys(@)[0]' --format='int'
+  FFLAGS      = $(_NODE_FMT) $(_IOTLAB_EXP_ID) $(_IOTLAB_NODELIST) --update $(IOTLAB_FLASHFILE) | $(_STDIN_EQ_0)
```

Comment:
Maybe rename this variable FLASH_FLAGS ?

Reply:
Do a `git grep FFLAGS` in RIOT. It is not mine.